app/controllers/WorkerController.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace GanttDashboard\App\Controllers;
+
+use GanttDashboard\App\Services\Worker as WorkerService;
+
+class WorkerController extends ControllerBase
+{
+    /**
+     * @var WorkerService
+     */
+    private $workerService;
+
+    /**
+     * Initialises the workerService property
+     */
+    public function onConstruct()
+    {
+        $this->workerService = $this->getDi()->get(WorkerService::class);
+    }
+
+    /**
+     * Performs the login
+     * @param string $accessKey
+     */
+    public function loginAction(string $accessKey = '')
```

Comment:
the logic in this method does not look right. please refactor or

Reply:
Hi. I understood that the login page must be hidden, so if the url doesn't match an valid url for login, the visitor should not be able to realize that there is a login action in worker controller. That is why I used this logic.